resources/product.py

```diff
@@ -40,66 +44,87 @@
 class Product(Resource):
     """CRUD for accessing/manipulating a single product document"""
 
-    @staticmethod
-    def get(product_id):
+    # GET - http://127.0.0.1:5000/api/product/<string:product_id>
+    @classmethod
+    @exception_handler
+    def get(cls, product_id):
         """Handles get request for retrieving a single product"""
         product = product_model.objects(_id=product_id).first()
         if len(product) == 0:
             abort(HTTPStatus.NOT_FOUND, message="Product could not be created")
         return json.loads(product.to_json()), HTTPStatus.CREATED
 
-    @staticmethod
+    # POST - http://127.0.0.1:5000/api/product
+    @classmethod
     @exception_handler
     @authenticated
-    def post():
+    def post(cls):
         """Handles the post request and creates a new product"""
-        product = product_args.parse_args()
-        new_product = product_model(**product)
+        body = clean_arguments(product_args)
+        product_id = ObjectId()
+        new_product = product_model(**body, _id=product_id)
```

Comment:
I don't think it's necessary to explicitly set an `ObjectId` since mongo does it in the background, but if it doesn't cause us any problem then it's all good

Reply:
So I create my own so I could pass that in the group model.